Fix Durable Objects TTL example where long TTLs lead to infinite alar… (#22728)
cloudflare/cloudflare-docs: src/content/docs/durable-objects/examples/durable-object-ttl.mdx

A common feature request for Durable Objects is a Time To Live (TTL) for Durable Object instances. Durable Objects give developers the tools to implement a custom TTL in only a few lines of code. This example demonstrates how to implement a TTL making use of <GlossaryTooltip term="alarm">`alarms`</GlossaryTooltip>. While this TTL will be extended upon every new request to the Durable Object, this can be customized based on a particular use case.

> :::caution[Not available in local development]

Reviewer: Why does the title say `not available in local dev`?

Reviewer 2: I think alarm support in local dev is limited/non-existent.

Author: This was a copy/paste error. Thank you for catching @vy-ton.

> The TTL is extended upon every new fetch request to the Durable Object. It might be tempting to instead extend the TTL in the constructor of the Durable Object. This is not advised because the Durable Object's constructor will be called before invoking the alarm handler if the alarm wakes the Durable Object up from hibernation. This approach will naively result in the constructor continually extending the TTL without running the alarm handler.

Reviewer 3: ```suggestion
The TTL is extended upon every new fetch request to the Durable Object. Cloudflare does not advise extending the TTL in the constructor of the Durable Object. This is because the Durable Object's constructor will be called before invoking the [alarm handler](/durable-objects/api/alarms/#handler-methods) if the alarm wakes the Durable Object from hibernation. This approach will naively result in the constructor continually extending the TTL without running the alarm handler.
```
